server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1258,6 +1258,10 @@ private void addStaleDocs(final List<LuceneDocument> docs, final IndexWriter ind
         }
     }
 
+    public long getRelativeTimeInNanos() {
```

Comment:
It looks like we have two different ways to inject the (fake) clock for the same purpose, could we instead pass down the time-supplier from `IndexShard` through the `EngineConfig`?